starbucks_dw/macros/calculate_response_rate.sql

```diff
@@ -0,0 +1,6 @@
+{% macro calculate_response_rate(transaction_status, offer_id) %}

```

Comment:
I would suggest a simplification of this calculation by a similar way:

```suggestion
{% macro calculate_response_rate(transaction_status_col, transaction_status='completed', offer_id_col) %}
   coalesce(count(*) filter (where {{ transaction_status_col }} = '{{ transaction_status }}' )
   / count(distinct {{ offer_id_col }} ), 0)
{% endmacro %}
```